pkg/admin/interface.go

```diff
@@ -228,9 +235,22 @@ func NewPulsarAdmin(conf PulsarAdminConfig) (PulsarAdmin, error) {
 		if err != nil {
 			return nil, err
 		}
-	} else {
+	} else if conf.Token != "" {
 		config.Token = conf.Token
 
+		adminClient, err = admin.New(config)
+		if err != nil {
+			return nil, err
+		}
+	} else if conf.ClientCertificatePath != "" {
+		config.AuthPlugin = auth.TLSPluginName
+		config.AuthParams = fmt.Sprintf("{\"tlsCertFile\": %q, \"tlsKeyFile\": %q}", conf.ClientCertificatePath, conf.ClientCertificateKeyPath)
```

Comment:
I didn't know about the `%q`.  I'll certainly be using that in the future.  If you want to go a little further and remove the escape slashes you could do. 
```suggestion
		config.AuthParams = fmt.Sprintf(`{"tlsCertFile": %q, "tlsKeyFile": %q}`, conf.ClientCertificatePath, conf.ClientCertificateKeyPath)
```